### ISSUE TITLE
[Test Failure] Failed to install package ‘Newtonsoft.Json’ into second C# Class Library App with ‘PackageReference’ mode in PM UI

### ISSUE THREAD

**Issue by v-luzh** [NuGet/Home#4654]
**NuGet Version:** NuGet 4.0.0-RTM-2300
**VS Version:** D15Rel 26217.02
OS Version: Windows 10 RS1_Release 14393.851.170215-1802

**Repro Steps:**
0. Check **‘Allow format selection on first package install’** format in NuGet Package Manager setting dialog.
1. Restart VS and create a C# Class Library (.NET Framework)-Project1.
2. Add another C# Class Library (.NET Framework)-Project2 into this solution.
3. Open the ‘Manage NuGet Packages…’ for project1.
4. Install package ‘Newtonsoft.Json’ into Project1 by PM UI with **PackageReference** mode.
5. Open the ‘Manage NuGet Packages…’ for project2.
6. Try to install package ‘Newtonsoft.Json’ into Project2 by PM UI **PackageReference** mode.

**Expected Result:**
Succeed to install package ‘Newtonsoft.Json’ into second C# Class Library.

**Actual Result:**
Failed to install package ‘Newtonsoft.Json’ into second C# Class Library with an error ‘Value cannot be null or an empty string.’ as below screenshot.
![twocl](https://cloud.githubusercontent.com/assets/20632625/23120153/35e76d22-f796-11e6-8e41-b6b38e0c0297.PNG)

**Note:**
1. It doesn’t repro when installing the same package in PMC.
2. It doesn’t repro when installing the same package with Packages.config mode in PM UI.
3. It repro on D15Rel 26210.02 (NuGet Client 4.0.0-RTM-2275)-V2 VSIX.

**Comment by alpaix** [comment]
Was able to repro. Call stack of the aforementioned exception:
```
System.ArgumentException: Value cannot be null or an empty string.
Parameter name: name
   at NuGet.PackageManagement.VisualStudio.ProjectSystemCache.ContainsKey(String name)
   at NuGet.PackageManagement.VisualStudio.VSSolutionManager.RemoveEnvDTEProjectFromCache(String name)
   at NuGet.PackageManagement.VisualStudio.VSSolutionManager.<UpdateNuGetProjectToPackageRef>d__61.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.PackageManagement.UI.NuGetUI.<UpdateNuGetProjectToPackageRef>d__9.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.PackageManagement.UI.UIActionEngine.<CheckPackageManagementFormat>d__8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at NuGet.PackageManagement.UI.UIActionEngine.<PerformActionImplAsync>d__7.MoveNext()
```

**Comment by alpaix** [comment]
Given the unusual usage pattern (opening two package management UIs simultaneously) and a simple workaround (re-opening the dialog would resolve the issue), punting to 4.0.1

**Comment by alpaix** [comment]
Also noted after package is installed into the first project it appears as installed in the second until the package management UI is re-opened. Seems like a different issue though.

**Comment by v-luzh** [comment]
@alpaix, you are right. That is another issue since they have different repro steps. I have update the repro steps on this one and filed a new bug #4675 to track that issue.

**Comment by v-luzh** [comment]
Fixed on **_VSUMain 26313.00+NuGet Client 4.3.0-Beta1-2381_** and **_D15Rel 26228.00+NuGet Client 4.1.0-RTM-2382_**.

**Comment by alpaix** [comment]
Confirmed issue has been resolved. Most likely with NuGet/NuGet.Client@5ef411e.
